### ISSUE TITLE
Blank nodes output as IRI in fromRdf

### ISSUE THREAD

**Issue by jmvanel** [filip26/titanium-json-ld#118]
**Describe the bug**
After building a single triple RdfDataset from API , and giving it to fromRdf, the blank node "bn1" is output as an IRI : 
`[{"@id":"urn:s1","urn:p1":[{"@id":"bn1"}]}]`

**To Reproduce**
This is Scala. I can provide a Java code, but I think you can manage that :) .

```scala
import com.apicatalog.rdf.impl.DefaultRdfProvider
import com.apicatalog.jsonld.api.impl.FromRdfApi
import com.apicatalog.jsonld.JsonLd
import com.apicatalog.jsonld.document.RdfDocument
import com.apicatalog.rdf.RdfDataset
import javax.json.JsonArray
import java.io.StringWriter
import javax.json.Json

object TestTitaniumBlankNodes extends App {
  val rdfProvider = DefaultRdfProvider.INSTANCE
  val subject = rdfProvider.createIRI("urn:s1")
  val objet = rdfProvider.createBlankNode("bn1")
  val predicate = rdfProvider.createIRI("urn:p1")
  val titaniumDS: RdfDataset = rdfProvider.createDataset()
  val nquad = rdfProvider.createNQuad(subject, predicate, objet, null)
  titaniumDS.add(nquad)

  val fromRdf: FromRdfApi =
      JsonLd.fromRdf(
        RdfDocument.of(titaniumDS) )

  printJsonArray(fromRdf . get) // KO: blank node as IRI !!!!!!!!!!!!!!!!!!!!

  def printJsonArray(jsa: JsonArray) {
    val sw = new StringWriter()
    val jsonWriter = Json.createWriter(sw)
    jsonWriter.writeArray(jsa)
    jsonWriter.close()
    sw.close()
    println(sw.toString())
  }
}
```
**Expected behavior**
Real blank node:
```json
[{
"@id":"urn:s1",
"urn:p1": "_:bn1" }]`
```
**Additional context**
Distilled from issue #89

**Comment by filip26** [comment]
Hi,
thank you for reporting it. `Rdf.createBlankNode()` method expected valid blank node notation without checking if the input is in valid notation or not. Now, the code you have provided will work and "_:" prefix is automatically added if missing. See commit [539f95a](https://github.com/filip26/titanium-json-ld/commit/539f59ac10d06d5192af2606d8220b21a760efc5#diff-709c22230c558cc20f7608bdc043afa61cd11495789d63f471c479401a5e2fe0R107)

Did you mean this as the expected output?

```json
[
    {
        "@id": "urn:s1",
        "urn:p1": [
            {
                "@id": "_:bn1"
            }
        ]
    }
]
````

**Comment by jmvanel** [comment]
Refering to the specif. 
https://w3c.github.io/json-ld-syntax/#identifying-blank-nodes
it is 
```json
[ {
        "@id": "urn:s1",
        "urn:p1": {
                "@id": "_:bn1"
         }
} ]
```
so, with `@id` , but without [] .

Indeed, my original proposition
```json
[{
"@id":"urn:s1",
"urn:p1": "_:bn1" 
}]
```
was wrong; checked with `$JENA/bin/riot blank.jsonld` , it gives a literal.

**Comment by filip26** [comment]
fromRdf method produces expanded form that includes JSON arrays. See https://w3c.github.io/json-ld-syntax/#identifying-blank-nodes - Expanded (result). I mean if you insist on 

```json
[ {
        "@id": "urn:s1",
        "urn:p1": {
                "@id": "_:bn1"
         }
} ]
```

then it should be treated as a separate bug, this issue will be closed by PR #119.

**Comment by jmvanel** [comment]
Thanks for he clarification; it's OK like this . We can close the issue !!!

I must confess that I learn JSON-LD partially by watching JSON-LD engines' output.
